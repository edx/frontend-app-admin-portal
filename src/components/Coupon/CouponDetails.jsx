import React from 'react';
import { Button, CheckBox, Icon } from '@edx/paragon';

import H3 from '../H3';

import TableContainer from '../../containers/TableContainer';
import DownloadCsvButton from '../../containers/DownloadCsvButton';
import EcommerceApiService from '../../data/services/EcommerceApiService';

class CouponDetails extends React.Component {
  constructor(props) {
    super(props);

    this.tableColumns = [
      {
        label: <CheckBox />,
        key: 'select',
      },
      {
        label: 'Assigned To',
        key: 'assigned_to',
      },
      {
        label: 'Redemptions',
        key: 'redemptions',
      },
      {
        label: 'Code',
        key: 'title',
      },
      {
        label: 'Actions',
        key: 'actions',
      },
    ];

    this.state = {
      selectedFilter: null,
    };

    this.formatCouponData = this.formatCouponData.bind(this);
  }

  setSelectedFilter(selectedFilter) {
    this.setState({
      selectedFilter,
    });
  }

  getActionButton(code) {
    const { assigned_to: assignedTo, redemptions } = code;

    if (redemptions.used === redemptions.available) {
      return null;
    }

    const button = {
      label: 'Assign',
      onClick: () => {},
    };

    if (assignedTo) {
      button.label = 'Revoke';
      button.onClick = () => {};
    }

    return (
      <Button
        className={[
          'btn-link',
          'btn-sm',
          'pl-0',
        ]}
        label={button.label}
        onClick={button.onClick}
      />
    );
  }

  formatCouponData(data) {
    return data.map(code => ({
      ...code,
      redemptions: `${code.redemptions.used} of ${code.redemptions.available}`,
      actions: this.getActionButton(code),
      select: <CheckBox />,
    }));
  }

  render() {
    const { selectedFilter } = this.state;

    return (
      <div className="details row no-gutters px-2 my-3">
        <div className="col">
          <div className="details-header row mb-2">
            <div className="col-12 col-md-6 mb-2 mb-md-0">
              <H3>Coupon Details</H3>
            </div>
            <div className="col-12 col-md-6 mb-2 mb-md-0 text-md-right">
              <DownloadCsvButton id="coupon-details" fetchMethod={() => {}} />
            </div>
          </div>
          <div className="row mb-3">
            <div className="col d-flex justify-content-between align-items-center">
              <div className="filters d-flex align-items-center">
                <span className="mr-2">Filter:</span>
                <Button
                  className={['btn-sm']}
                  buttonType={selectedFilter === 'not-assigned' ? 'primary' : 'link'}
                  label="Not Assigned"
                  onClick={() => this.setSelectedFilter('not-assigned')}
                />
                <Button
                  className={['btn-sm']}
                  buttonType={selectedFilter === 'not-redeemed' ? 'primary' : 'link'}
                  label="Not Redeemed"
                  onClick={() => this.setSelectedFilter('not-redeemed')}
                />
                {selectedFilter &&
                  <Button
                    className={['btn-sm', 'btn-outline-primary', 'ml-2']}
                    label={
                      <React.Fragment>
                        <Icon className={['fa', 'fa-close', 'mr-1']} />
                        Clear filters
                      </React.Fragment>
                    }
                    onClick={() => this.setSelectedFilter(null)}
                  />
                }
              </div>
            </div>
          </div>
          <TableContainer
            id="coupon-details"
            className="coupon-details"
            fetchMethod={EcommerceApiService.fetchCouponDetails}
            columns={this.tableColumns}
            formatData={this.formatCouponData}
          />
        </div>
      </div>
    );
  }
}

export default CouponDetails;
