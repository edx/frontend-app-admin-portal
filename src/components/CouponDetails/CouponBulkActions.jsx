import React from 'react';
import PropTypes from 'prop-types';

import {
  Button, Form,
} from '@edx/paragon';
import { BULK_ACTION } from './constants';

const CouponBulkActions = ({
  value, options, disabled, onChange, handleBulkAction,
}) => (
  <>
    <Form.Group className="flex-grow-1" controlId={BULK_ACTION.controlId} name={BULK_ACTION.name}>
      <Form.Control
        className="float-right"
        floatingLabel={BULK_ACTION.label}
        as="select"
        value={value}
        onChange={(e) => onChange(e.target.value)}
        disabled={disabled}
      >
        {options.map(
          ({ label, value: optionValue, disabled: optionDisabled }) => (
            <option
              key={optionValue}
              value={optionValue}
              disabled={optionDisabled}
            >
              {label}
            </option>
          ),
        )}
      </Form.Control>
    </Form.Group>
    <Button
      className="ml-2 mb-3"
      onClick={handleBulkAction}
      disabled={disabled}
    >
      Go
    </Button>
  </>
);

CouponBulkActions.propTypes = {
  value: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    disabled: PropTypes.bool.isRequired,
  })).isRequired,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired,
  handleBulkAction: PropTypes.func.isRequired,
};

export default CouponBulkActions;
