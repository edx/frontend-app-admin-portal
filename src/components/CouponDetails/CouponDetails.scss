@import "~bootstrap/scss/functions";
@import "~bootstrap/scss/variables";

.coupon-details {
    font-size: 1rem;

    .loading {
        font-size: inherit;
    }

    .alert {
        margin-bottom: 0;
        font-size: 0.85rem;

        .fa {
            font-size: 1rem;
        }
    }

    .table,
    .pagination .btn {
        font-size: 0.85rem;
    }

    .table {
        td {
            vertical-align: middle;
        }

        .btn {
            font-size: inherit;
            text-decoration: underline;

            &:focus {
                box-shadow: $btn-focus-box-shadow;
            }
        }
    }
    
    .toggles,
    .bulk-actions {
        .form-group {
            display: inline-block;

            &,
            label {
                margin: 0;
            }

            label {
                font-weight: 600;
            }
        }
    }
}
