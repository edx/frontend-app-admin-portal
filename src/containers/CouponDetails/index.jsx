import { connect } from 'react-redux';

import CouponDetails from '../../components/CouponDetails';
import { fetchCouponOrder } from '../../data/actions/coupons';

const mapStateToProps = state => ({
  couponDetailsTable: state.table['coupon-details'],
  couponOverviewError: state.coupons.couponOverviewError,
  couponOverviewLoading: state.coupons.couponOverviewLoading,
});

const mapDispatchToProps = dispatch => ({
  fetchCouponOrder: (couponId) => {
    dispatch(fetchCouponOrder(couponId));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(CouponDetails);
