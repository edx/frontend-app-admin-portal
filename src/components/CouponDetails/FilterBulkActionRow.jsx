import React from 'react';
import PropTypes from 'prop-types';

import CouponFilters from './CouponFilters';
import CouponBulkActions from './CouponBulkActions';
import { COUPON_FILTER_TYPES } from './constants';

// Display component to handle layout

const FilterBulkActionRow = ({ couponFilterProps, couponBulkActionProps, ...sharedProps }) => (
  <div className="d-flex justify-content-between mt-3">
    <div className="toggles">
      <CouponFilters
        {...couponFilterProps}
        {...sharedProps}
      />
    </div>
    <div className="bulk-actions m-md-0 d-flex justify-content-end">
      <CouponBulkActions
        {...couponBulkActionProps}
        {...sharedProps}
      />
    </div>
  </div>
);

FilterBulkActionRow.propTypes = {
  // specific PropTypes are defined on the CouponFilters and CouponBulkActions components, respectively
  couponFilterProps: PropTypes.shape().isRequired,
  couponBulkActionProps: PropTypes.shape().isRequired,
  isTableLoading: PropTypes.bool.isRequired,
  selectedToggle: PropTypes.oneOf(Object.values(COUPON_FILTER_TYPES)).isRequired,

};

export default FilterBulkActionRow;
