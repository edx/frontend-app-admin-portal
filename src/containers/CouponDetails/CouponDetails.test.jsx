import React from 'react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';

import CouponDetails from './index';

const mockStore = configureMockStore([thunk]);
const initialState = {
  csv: {
    'coupon-details': {},
  },
  table: {
    'coupon-details': {},
  },
};

const CouponDetailsWrapper = props => (
  <MemoryRouter>
    <Provider store={props.store}>
      <CouponDetails
        id={1}
        couponTitle="test-title"
        {...props}
      />
    </Provider>
  </MemoryRouter>
);

CouponDetailsWrapper.defaultProps = {
  store: mockStore({ ...initialState }),
};

CouponDetailsWrapper.propTypes = {
  store: PropTypes.shape({}),
};

const sampleCodeData = {
  code: 'test-code-1',
  assigned_to: 'test@bestrun.com',
  redemptions: {
    available: 100,
    used: 10,
  },
  error: null,
};

const sampleTableData = {
  loading: false,
  error: null,
  data: {
    count: 5,
    num_pages: 2,
    current_page: 1,
    results: [
      sampleCodeData,
      {
        ...sampleCodeData,
        code: 'test-code-2',
        redemptions: {
          available: 100,
          used: 100,
        },
      },
      {
        ...sampleCodeData,
        code: 'test-code-3',
        assigned_to: null,
      },
    ],
  },
};

describe('CouponDetailsWrapper', () => {
  describe('renders', () => {
    it('with collapsed coupon details', () => {
      const tree = renderer
        .create((
          <CouponDetailsWrapper />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with expanded coupon details', () => {
      const tree = renderer
        .create((
          <CouponDetailsWrapper isExpanded />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with error', () => {
      const tree = renderer
        .create((
          <CouponDetailsWrapper hasError isExpanded />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with table data', () => {
      const store = mockStore({
        ...initialState,
        table: {
          'coupon-details': sampleTableData,
        },
      });
      const tree = renderer
        .create((
          <CouponDetailsWrapper
            store={store}
            isExpanded
          />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('handles isExpanded prop change', () => {
    const wrapper = mount(<CouponDetailsWrapper isExpanded />);
    expect(wrapper.find(CouponDetails).prop('isExpanded')).toBeTruthy();

    wrapper.setProps({
      isExpanded: false,
    });
    expect(wrapper.find(CouponDetails).prop('isExpanded')).toBeFalsy();

    wrapper.setProps({
      isExpanded: true,
    });
    expect(wrapper.find(CouponDetails).prop('isExpanded')).toBeTruthy();
  });

  it('properly handles changes to selected toggle input', () => {
    const wrapper = mount(<CouponDetailsWrapper isExpanded />);
    expect(wrapper.find('select').first().prop('value')).toEqual('unassigned');

    wrapper.find('select').first().simulate('change', { target: { value: 'redeemed' } });
    expect(wrapper.find('select').first().prop('value')).toEqual('redeemed');

    wrapper.find('select').first().simulate('change', { target: { value: 'unassigned' } });
    expect(wrapper.find('select').first().prop('value')).toEqual('unassigned');
  });

  describe('modals', () => {
    let store;
    let wrapper;

    const openModalByActionButton = ({ key, label }) => {
      const buttons = wrapper.find('table').find('button');
      const actionButton = key === 'revoke' ? buttons.first() : buttons.last();
      expect(actionButton.prop('children')).toEqual(label);
      actionButton.simulate('click');
      expect(wrapper.find('CouponDetails').instance().state.modals[key]).toBeTruthy();
    };

    beforeEach(() => {
      store = mockStore({
        ...initialState,
        table: {
          'coupon-details': sampleTableData,
        },
      });

      wrapper = mount((
        <CouponDetailsWrapper
          store={store}
          isExpanded
        />
      ));
    });

    it('sets revoke modal state on Revoke button click', () => {
      openModalByActionButton({
        key: 'revoke',
        label: 'Revoke',
      });
    });

    it('sets assignment modal state on Assign button click', () => {
      openModalByActionButton({
        key: 'assignment',
        label: 'Assign',
      });
    });

    it('sets revoke modal state on bulk revoke click', () => {
      wrapper.find('.toggles select').simulate('change', { target: { value: 'unredeemed' } });
      expect(wrapper.find('.toggles select').prop('value')).toEqual('unredeemed');

      wrapper.find('.bulk-actions select').simulate('change', { target: { value: 'revoke' } });
      expect(wrapper.find('.bulk-actions select').prop('value')).toEqual('revoke');

      wrapper.find('.bulk-actions .btn').simulate('click');
      expect(wrapper.find('CouponDetails').instance().state.modals.revoke).toBeTruthy();
    });

    it('sets assignment modal state on bulk assign click', () => {
      wrapper.find('.bulk-actions .btn').simulate('click');
      expect(wrapper.find('CouponDetails').instance().state.modals.assignment).toBeTruthy();
    });

    it('does not open modal on unknown bulk action', () => {
      wrapper.find('.bulk-actions select').simulate('change', { target: { value: 'test' } });
      wrapper.find('.bulk-actions .btn').simulate('click');

      expect(wrapper.find('CouponDetails').instance().state.modals.assignment).toBeFalsy();
      expect(wrapper.find('CouponDetails').instance().state.modals.revoke).toBeFalsy();
    });

    it('handles successful code assignment from modal', () => {
      openModalByActionButton({
        key: 'assignment',
        label: 'Assign',
      });

      // fake successful code assignment
      wrapper.find('CodeAssignmentModal').prop('onSuccess')();
      expect(wrapper.find('CouponDetails').instance().state.isCodeAssignmentSuccessful).toBeTruthy();

      wrapper.update();

      // success status alert
      expect(wrapper.find('StatusAlert').prop('alertType')).toEqual('success');
      wrapper.find('StatusAlert').find('.alert-dialog .btn').simulate('click');

      expect(wrapper.find('CouponDetails').instance().state.isCodeAssignmentSuccessful).toBeFalsy();
      expect(wrapper.find('CouponDetails').instance().state.selectedToggle).toEqual('unredeemed');
    });
  });

  describe('code selection', () => {
    let store;
    let wrapper;

    beforeEach(() => {
      store = mockStore({
        ...initialState,
        table: {
          'coupon-details': sampleTableData,
        },
      });
      wrapper = mount((
        <CouponDetailsWrapper
          store={store}
          isExpanded
        />
      ));
    });

    const selectAllCodesOnPage = ({ isSelected, expectedSelectionLength }) => {
      const selectAllCheckbox = wrapper.find('table th').find('input[type=\'checkbox\']');
      selectAllCheckbox.simulate('change', { target: { value: isSelected } });
      expect(wrapper.find('CouponDetails').instance().state.selectedCodes).toHaveLength(expectedSelectionLength);
    };

    it('handles individual code selection within table', () => {
      const checkboxes = wrapper.find('table').find('input[type=\'checkbox\']').slice(1);

      checkboxes.first().simulate('change', { target: { value: true } });
      checkboxes.last().simulate('change', { target: { value: true } });

      expect(wrapper.find('CouponDetails').instance().state.selectedCodes).toHaveLength(2);

      checkboxes.first().simulate('change', { target: { value: false } });
      expect(wrapper.find('CouponDetails').instance().state.selectedCodes).toHaveLength(1);
    });

    it('handles select all checkbox click', () => {
      // select all codes
      selectAllCodesOnPage({
        isSelected: true,
        expectedSelectionLength: 3,
      });

      // unselect all codes
      selectAllCodesOnPage({
        isSelected: false,
        expectedSelectionLength: 0,
      });
    });

    it('handles select all codes across pages', () => {
      selectAllCodesOnPage({
        isSelected: true,
        expectedSelectionLength: 3,
      });

      const alert = wrapper.find('CouponDetails').find('.alert');
      expect(alert).toBeTruthy();
      alert.find('.btn').simulate('click');

      expect(wrapper.find('CouponDetails').instance().state.hasAllCodesSelected).toBeTruthy();
    });
  });
});
