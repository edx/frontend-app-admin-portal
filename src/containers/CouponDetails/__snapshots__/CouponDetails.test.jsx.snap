// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CouponDetailsWrapper renders with collapsed coupon details 1`] = `
<div
  className="coupon-details row no-gutters px-2 my-3 d-none"
  id="coupon-details-1"
>
  <div
    className="col"
  />
</div>
`;

exports[`CouponDetailsWrapper renders with error 1`] = `
<div
  className="coupon-details row no-gutters px-2 my-3"
  id="coupon-details-1"
>
  <div
    className="col"
  >
    <div
      className="details-header row no-gutters mb-3"
    >
      <div
        className="col-12 col-md-6 mb-2 mb-md-0"
      >
        <h3
          className={null}
        >
          Coupon Details
        </h3>
      </div>
      <div
        className="col-12 col-md-6 mb-2 mb-md-0 text-md-right"
      >
        <button
          className="btn btn-outline-primary download-btn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="fa mr-2 fa-download"
            id="Icon1"
          />
          Download full report (CSV)
        </button>
      </div>
    </div>
    <div
      className="row mb-3"
    >
      <div
        className="toggles col-12 col-md-8"
      >
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="asInput5"
            id="label-asInput5"
          >
            Filter by Code Status:
          </label>
          <select
            aria-describedby="error-asInput5"
            className="form-control is-invalid-nodanger mt-1"
            disabled={false}
            id="asInput5"
            name="table-view"
            onBlur={[Function]}
            onChange={[Function]}
            type="select"
            value="unassigned"
          >
            <option
              value="unassigned"
            >
              Unassigned
            </option>
            <option
              value="unredeemed"
            >
              Unredeemed
            </option>
            <option
              value="partially-redeemed"
            >
              Partially Redeemed
            </option>
            <option
              value="redeemed"
            >
              Redeemed
            </option>
          </select>
          <div
            aria-live="polite"
            className="invalid-feedback invalid-feedback-nodanger"
            id="error-asInput5"
          >
            <span />
          </div>
        </div>
      </div>
      <div
        className="bulk-actions col-12 col-md-4 text-md-right mt-3 m-md-0"
      >
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="asInput6"
            id="label-asInput6"
          >
            Bulk Action:
          </label>
          <select
            aria-describedby="error-asInput6"
            className="form-control is-invalid-nodanger mt-1"
            disabled={false}
            id="asInput6"
            name="bulk-action"
            onBlur={[Function]}
            onChange={[Function]}
            type="select"
            value="assign"
          >
            <option
              disabled={false}
              value="assign"
            >
              Assign
            </option>
            <option
              disabled={true}
              value="remind"
            >
              Remind
            </option>
            <option
              disabled={true}
              value="revoke"
            >
              Revoke
            </option>
          </select>
          <div
            aria-live="polite"
            className="invalid-feedback invalid-feedback-nodanger"
            id="error-asInput6"
          >
            <span />
          </div>
        </div>
        <button
          className="btn ml-2 btn-primary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          Go
        </button>
      </div>
    </div>
    <div
      className="row mb-3"
    >
      <div
        className="col"
      >
        <div
          className="alert fade alert-danger show"
          hidden={false}
          role="alert"
        >
          <div
            className="alert-dialog"
          >
            <div
              className="d-flex"
            >
              <div
                className="icon mr-2"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-times-circle"
                  id="Icon1"
                />
              </div>
              <div
                className="message"
              >
                One or more codes below have an error.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CouponDetailsWrapper renders with expanded coupon details 1`] = `
<div
  className="coupon-details row no-gutters px-2 my-3"
  id="coupon-details-1"
>
  <div
    className="col"
  >
    <div
      className="details-header row no-gutters mb-3"
    >
      <div
        className="col-12 col-md-6 mb-2 mb-md-0"
      >
        <h3
          className={null}
        >
          Coupon Details
        </h3>
      </div>
      <div
        className="col-12 col-md-6 mb-2 mb-md-0 text-md-right"
      >
        <button
          className="btn btn-outline-primary download-btn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="fa mr-2 fa-download"
            id="Icon1"
          />
          Download full report (CSV)
        </button>
      </div>
    </div>
    <div
      className="row mb-3"
    >
      <div
        className="toggles col-12 col-md-8"
      >
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="asInput3"
            id="label-asInput3"
          >
            Filter by Code Status:
          </label>
          <select
            aria-describedby="error-asInput3"
            className="form-control is-invalid-nodanger mt-1"
            disabled={false}
            id="asInput3"
            name="table-view"
            onBlur={[Function]}
            onChange={[Function]}
            type="select"
            value="unassigned"
          >
            <option
              value="unassigned"
            >
              Unassigned
            </option>
            <option
              value="unredeemed"
            >
              Unredeemed
            </option>
            <option
              value="partially-redeemed"
            >
              Partially Redeemed
            </option>
            <option
              value="redeemed"
            >
              Redeemed
            </option>
          </select>
          <div
            aria-live="polite"
            className="invalid-feedback invalid-feedback-nodanger"
            id="error-asInput3"
          >
            <span />
          </div>
        </div>
      </div>
      <div
        className="bulk-actions col-12 col-md-4 text-md-right mt-3 m-md-0"
      >
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="asInput4"
            id="label-asInput4"
          >
            Bulk Action:
          </label>
          <select
            aria-describedby="error-asInput4"
            className="form-control is-invalid-nodanger mt-1"
            disabled={false}
            id="asInput4"
            name="bulk-action"
            onBlur={[Function]}
            onChange={[Function]}
            type="select"
            value="assign"
          >
            <option
              disabled={false}
              value="assign"
            >
              Assign
            </option>
            <option
              disabled={true}
              value="remind"
            >
              Remind
            </option>
            <option
              disabled={true}
              value="revoke"
            >
              Revoke
            </option>
          </select>
          <div
            aria-live="polite"
            className="invalid-feedback invalid-feedback-nodanger"
            id="error-asInput4"
          >
            <span />
          </div>
        </div>
        <button
          className="btn ml-2 btn-primary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          Go
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CouponDetailsWrapper renders with table data 1`] = `
<div
  className="coupon-details row no-gutters px-2 my-3"
  id="coupon-details-1"
>
  <div
    className="col"
  >
    <div
      className="details-header row no-gutters mb-3"
    >
      <div
        className="col-12 col-md-6 mb-2 mb-md-0"
      >
        <h3
          className={null}
        >
          Coupon Details
        </h3>
      </div>
      <div
        className="col-12 col-md-6 mb-2 mb-md-0 text-md-right"
      >
        <button
          className="btn btn-outline-primary download-btn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="fa mr-2 fa-download"
            id="Icon1"
          />
          Download full report (CSV)
        </button>
      </div>
    </div>
    <div
      className="row mb-3"
    >
      <div
        className="toggles col-12 col-md-8"
      >
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="asInput7"
            id="label-asInput7"
          >
            Filter by Code Status:
          </label>
          <select
            aria-describedby="error-asInput7"
            className="form-control is-invalid-nodanger mt-1"
            disabled={false}
            id="asInput7"
            name="table-view"
            onBlur={[Function]}
            onChange={[Function]}
            type="select"
            value="unassigned"
          >
            <option
              value="unassigned"
            >
              Unassigned
            </option>
            <option
              value="unredeemed"
            >
              Unredeemed
            </option>
            <option
              value="partially-redeemed"
            >
              Partially Redeemed
            </option>
            <option
              value="redeemed"
            >
              Redeemed
            </option>
          </select>
          <div
            aria-live="polite"
            className="invalid-feedback invalid-feedback-nodanger"
            id="error-asInput7"
          >
            <span />
          </div>
        </div>
      </div>
      <div
        className="bulk-actions col-12 col-md-4 text-md-right mt-3 m-md-0"
      >
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="asInput8"
            id="label-asInput8"
          >
            Bulk Action:
          </label>
          <select
            aria-describedby="error-asInput8"
            className="form-control is-invalid-nodanger mt-1"
            disabled={false}
            id="asInput8"
            name="bulk-action"
            onBlur={[Function]}
            onChange={[Function]}
            type="select"
            value="assign"
          >
            <option
              disabled={false}
              value="assign"
            >
              Assign
            </option>
            <option
              disabled={true}
              value="remind"
            >
              Remind
            </option>
            <option
              disabled={true}
              value="revoke"
            >
              Revoke
            </option>
          </select>
          <div
            aria-live="polite"
            className="invalid-feedback invalid-feedback-nodanger"
            id="error-asInput8"
          >
            <span />
          </div>
        </div>
        <button
          className="btn ml-2 btn-primary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          Go
        </button>
      </div>
    </div>
    <div
      className="coupon-details-table"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="table-responsive"
          >
            <table
              className="table table-sm table-striped"
            >
              <thead
                className=""
              >
                <tr
                  className=""
                >
                  <th
                    className=""
                    scope="col"
                  >
                    <div
                      className="form-check"
                    >
                      <input
                        aria-checked={false}
                        checked={false}
                        className={
                          Array [
                            "form-check-input is-invalid-nodanger",
                          ]
                        }
                        disabled={false}
                        id="asInput9"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <label
                        className="form-check-label"
                        htmlFor="asInput9"
                        id="label-asInput9"
                      />
                      <div
                        aria-live="polite"
                        className="invalid-feedback invalid-feedback-nodanger"
                        id="error-asInput9"
                      >
                        <span />
                      </div>
                    </div>
                  </th>
                  <th
                    className=""
                    scope="col"
                  >
                    Redemptions
                  </th>
                  <th
                    className=""
                    scope="col"
                  >
                    Code
                  </th>
                  <th
                    className=""
                    scope="col"
                  >
                    Actions
                  </th>
                </tr>
              </thead>
              <tbody
                className=""
              >
                <tr
                  className=""
                >
                  <td
                    className=""
                  >
                    <div
                      className="form-check"
                    >
                      <input
                        aria-checked={false}
                        checked={false}
                        className={
                          Array [
                            "form-check-input is-invalid-nodanger",
                          ]
                        }
                        disabled={false}
                        id="asInput10"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <label
                        className="form-check-label"
                        htmlFor="asInput10"
                        id="label-asInput10"
                      />
                      <div
                        aria-live="polite"
                        className="invalid-feedback invalid-feedback-nodanger"
                        id="error-asInput10"
                      >
                        <span />
                      </div>
                    </div>
                  </td>
                  <td
                    className=""
                  >
                    10 of 100
                  </td>
                  <td
                    className=""
                  >
                    test-code-1
                  </td>
                  <td
                    className=""
                  >
                    <button
                      className="btn btn-link btn-sm px-0"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      Revoke
                    </button>
                  </td>
                </tr>
                <tr
                  className=""
                >
                  <td
                    className=""
                  >
                    <div
                      className="form-check"
                    >
                      <input
                        aria-checked={false}
                        checked={false}
                        className={
                          Array [
                            "form-check-input is-invalid-nodanger",
                          ]
                        }
                        disabled={false}
                        id="asInput11"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <label
                        className="form-check-label"
                        htmlFor="asInput11"
                        id="label-asInput11"
                      />
                      <div
                        aria-live="polite"
                        className="invalid-feedback invalid-feedback-nodanger"
                        id="error-asInput11"
                      >
                        <span />
                      </div>
                    </div>
                  </td>
                  <td
                    className=""
                  >
                    100 of 100
                  </td>
                  <td
                    className=""
                  >
                    test-code-2
                  </td>
                  <td
                    className=""
                  />
                </tr>
                <tr
                  className=""
                >
                  <td
                    className=""
                  >
                    <div
                      className="form-check"
                    >
                      <input
                        aria-checked={false}
                        checked={false}
                        className={
                          Array [
                            "form-check-input is-invalid-nodanger",
                          ]
                        }
                        disabled={false}
                        id="asInput12"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <label
                        className="form-check-label"
                        htmlFor="asInput12"
                        id="label-asInput12"
                      />
                      <div
                        aria-live="polite"
                        className="invalid-feedback invalid-feedback-nodanger"
                        id="error-asInput12"
                      >
                        <span />
                      </div>
                    </div>
                  </td>
                  <td
                    className=""
                  >
                    10 of 100
                  </td>
                  <td
                    className=""
                  >
                    test-code-3
                  </td>
                  <td
                    className=""
                  >
                    <button
                      className="btn btn-link btn-sm px-0"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      Assign
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        className="row mt-2"
      >
        <div
          className="col d-flex justify-content-center"
        >
          <nav
            aria-label="coupon-details-pagination"
            className=""
          >
            <div
              aria-atomic={true}
              aria-live="polite"
              aria-relevant="text"
              className="sr-only"
            >
              Page 1, Current Page, of 2
            </div>
            <ul
              className="pagination"
            >
              <li
                className="page-item disabled"
              >
                <button
                  aria-label="Previous"
                  className="btn previous page-link"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  tabIndex="-1"
                  type="button"
                >
                  <div>
                    <span
                      aria-hidden={true}
                      className="fa fa-chevron-left mr-2"
                      id="pagination-13"
                    />
                    Previous
                  </div>
                </button>
              </li>
              <li
                className="page-item"
              >
                <button
                  aria-label="Next, Page 2"
                  className="btn next page-link"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <div>
                    Next
                    <span
                      aria-hidden={true}
                      className="fa fa-chevron-right ml-2"
                      id="pagination-14"
                    />
                  </div>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>
</div>
`;
